Fix compilation warning for EXPO and VERSION

```shell
../src/forthcoming/expo/C/expot.c:102:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  102 |                                            fc, gj, hl );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const float *, float *, float *, const void *)}
In file included from ../src/forthcoming/expo/C/expot.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const float *, float,  float *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const float *, float *, float *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/forthcoming/expo/C/expot.c:112:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  112 |                                            fc, gj, hl );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const float *, float *, float *, const void *)}
In file included from ../src/forthcoming/expo/C/expot.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const float *, float,  float *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const float *, float *, float *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/forthcoming/expo/C/expot.c:122:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  122 |                                            fc, gj_dense, hl_dense );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const float *, float *, float *, const void *)}
In file included from ../src/forthcoming/expo/C/expot.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const float *, float,  float *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const float *, float *, float *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/forthcoming/expo/C/expot.c:132:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  132 |                                            fc, gj, hl );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const float *, float *, float *, const void *)}
In file included from ../src/forthcoming/expo/C/expot.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const float *, float,  float *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const float *, float *, float *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[3632/3966] Compiling C object expot_c_double.p/src_forthcoming_expo_C_expot.c.o
../src/forthcoming/expo/C/expot.c: In function ‘main’:
../src/forthcoming/expo/C/expot.c:102:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  102 |                                            fc, gj, hl );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const double *, double *, double *, const void *)}
In file included from ../src/forthcoming/expo/C/expot.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const double *, double,  double *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const double *, double *, double *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/forthcoming/expo/C/expot.c:112:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  112 |                                            fc, gj, hl );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const double *, double *, double *, const void *)}
In file included from ../src/forthcoming/expo/C/expot.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const double *, double,  double *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const double *, double *, double *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/forthcoming/expo/C/expot.c:122:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  122 |                                            fc, gj_dense, hl_dense );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const double *, double *, double *, const void *)}
In file included from ../src/forthcoming/expo/C/expot.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const double *, double,  double *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const double *, double *, double *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/forthcoming/expo/C/expot.c:132:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  132 |                                            fc, gj, hl );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const double *, double *, double *, const void *)}
In file included from ../src/forthcoming/expo/C/expot.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const double *, double,  double *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const double *, double *, double *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[3633/3966] Compiling C object expotf_c_single.p/src_forthcoming_expo_C_expotf.c.o
../src/forthcoming/expo/C/expotf.c: In function ‘main’:
../src/forthcoming/expo/C/expotf.c:101:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  101 |                                            fc, gj, hl );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const float *, float *, float *, const void *)}
In file included from ../src/forthcoming/expo/C/expotf.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const float *, float,  float *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const float *, float *, float *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/forthcoming/expo/C/expotf.c:111:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  111 |                                            fc, gj, hl );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const float *, float *, float *, const void *)}
In file included from ../src/forthcoming/expo/C/expotf.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const float *, float,  float *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const float *, float *, float *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/forthcoming/expo/C/expotf.c:121:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  121 |                                            fc, gj_dense, hl_dense );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const float *, float *, float *, const void *)}
In file included from ../src/forthcoming/expo/C/expotf.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const float *, float,  float *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const float *, float *, float *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/forthcoming/expo/C/expotf.c:131:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  131 |                                            fc, gj, hl );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const float *, float *, float *, const void *)}
In file included from ../src/forthcoming/expo/C/expotf.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const float *, float,  float *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const float *, float *, float *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[3634/3966] Compiling C object expotf_c_double.p/src_forthcoming_expo_C_expotf.c.o
../src/forthcoming/expo/C/expotf.c: In function ‘main’:
../src/forthcoming/expo/C/expotf.c:101:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  101 |                                            fc, gj, hl );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const double *, double *, double *, const void *)}
In file included from ../src/forthcoming/expo/C/expotf.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const double *, double,  double *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const double *, double *, double *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/forthcoming/expo/C/expotf.c:111:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  111 |                                            fc, gj, hl );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const double *, double *, double *, const void *)}
In file included from ../src/forthcoming/expo/C/expotf.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const double *, double,  double *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const double *, double *, double *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/forthcoming/expo/C/expotf.c:121:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  121 |                                            fc, gj_dense, hl_dense );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const double *, double *, double *, const void *)}
In file included from ../src/forthcoming/expo/C/expotf.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const double *, double,  double *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const double *, double *, double *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/forthcoming/expo/C/expotf.c:131:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  131 |                                            fc, gj, hl );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const double *, double *, double *, const void *)}
In file included from ../src/forthcoming/expo/C/expotf.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const double *, double,  double *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const double *, double *, double *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[3635/3966] Compiling C object expot_c_quadruple.p/src_forthcoming_expo_C_expot.c.o
../src/forthcoming/expo/C/expot.c: In function ‘main’:
../src/forthcoming/expo/C/expot.c:102:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  102 |                                            fc, gj, hl );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const _Float128 *, _Float128 *, _Float128 *, const void *)}
In file included from ../src/forthcoming/expo/C/expot.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const _Float128 *, _Float128,  _Float128 *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const _Float128 *, _Float128 *, _Float128 *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/forthcoming/expo/C/expot.c:112:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  112 |                                            fc, gj, hl );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const _Float128 *, _Float128 *, _Float128 *, const void *)}
In file included from ../src/forthcoming/expo/C/expot.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const _Float128 *, _Float128,  _Float128 *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const _Float128 *, _Float128 *, _Float128 *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/forthcoming/expo/C/expot.c:122:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  122 |                                            fc, gj_dense, hl_dense );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const _Float128 *, _Float128 *, _Float128 *, const void *)}
In file included from ../src/forthcoming/expo/C/expot.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const _Float128 *, _Float128,  _Float128 *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const _Float128 *, _Float128 *, _Float128 *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/forthcoming/expo/C/expot.c:132:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  132 |                                            fc, gj, hl );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const _Float128 *, _Float128 *, _Float128 *, const void *)}
In file included from ../src/forthcoming/expo/C/expot.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const _Float128 *, _Float128,  _Float128 *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const _Float128 *, _Float128 *, _Float128 *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[3637/3966] Compiling C object expotf_c_quadruple.p/src_forthcoming_expo_C_expotf.c.o
../src/forthcoming/expo/C/expotf.c: In function ‘main’:
../src/forthcoming/expo/C/expotf.c:101:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  101 |                                            fc, gj, hl );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const _Float128 *, _Float128 *, _Float128 *, const void *)}
In file included from ../src/forthcoming/expo/C/expotf.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const _Float128 *, _Float128,  _Float128 *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const _Float128 *, _Float128 *, _Float128 *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/forthcoming/expo/C/expotf.c:111:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  111 |                                            fc, gj, hl );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const _Float128 *, _Float128 *, _Float128 *, const void *)}
In file included from ../src/forthcoming/expo/C/expotf.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const _Float128 *, _Float128,  _Float128 *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const _Float128 *, _Float128 *, _Float128 *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/forthcoming/expo/C/expotf.c:121:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  121 |                                            fc, gj_dense, hl_dense );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const _Float128 *, _Float128 *, _Float128 *, const void *)}
In file included from ../src/forthcoming/expo/C/expotf.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const _Float128 *, _Float128,  _Float128 *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const _Float128 *, _Float128 *, _Float128 *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/forthcoming/expo/C/expotf.c:131:44: warning: passing argument 17 of ‘expo_solve_hessian_direct’ from incompatible pointer type [-Wincompatible-pointer-types]
  131 |                                            fc, gj, hl );
      |                                            ^~
      |                                            |
      |                                            ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *) {aka long int (*)(long int,  long int,  const _Float128 *, _Float128 *, _Float128 *, const void *)}
In file included from ../src/forthcoming/expo/C/expotf.c:9:
../include/galahad_expo.h:883:40: note: expected ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_,  rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const _Float128 *, _Float128,  _Float128 *, const void *)’} but argument is of type ‘ipc_ (*)(ipc_,  ipc_,  const rpc_ *, rpc_ *, rpc_ *, const void *)’ {aka ‘long int (*)(long int,  long int,  const _Float128 *, _Float128 *, _Float128 *, const void *)’}
  883 |                                 ipc_ (*eval_fc)(
      |                                 ~~~~~~~^~~~~~~~~
  884 |                                   ipc_, ipc_, const rpc_[],
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  885 |                                   rpc_, rpc_[], const void * ),
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
```shell
[3943/3966] Compiling C object versiont_c_single.p/src_version_C_versiont.c.o
../src/version/C/versiont.c: In function ‘main’:
../src/version/C/versiont.c:15:33: warning: format ‘%d’ expects argument of type ‘int’, but argument 2 has type ‘ipc_’ {aka ‘long int’} [-Wformat=]
   15 |     printf( " GALAHAD version: %d.%d.%d\n", major, minor, patch );
      |                                ~^           ~~~~~
      |                                 |           |
      |                                 int         ipc_ {aka long int}
      |                                %ld
../src/version/C/versiont.c:15:36: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘ipc_’ {aka ‘long int’} [-Wformat=]
   15 |     printf( " GALAHAD version: %d.%d.%d\n", major, minor, patch );
      |                                   ~^               ~~~~~
      |                                    |               |
      |                                    int             ipc_ {aka long int}
      |                                   %ld
../src/version/C/versiont.c:15:39: warning: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘ipc_’ {aka ‘long int’} [-Wformat=]
   15 |     printf( " GALAHAD version: %d.%d.%d\n", major, minor, patch );
      |                                      ~^                   ~~~~~
      |                                       |                   |
      |                                       int                 ipc_ {aka long int}
      |                                      %ld
[3948/3966] Compiling C object versiont_c_double.p/src_version_C_versiont.c.o
../src/version/C/versiont.c: In function ‘main’:
../src/version/C/versiont.c:15:33: warning: format ‘%d’ expects argument of type ‘int’, but argument 2 has type ‘ipc_’ {aka ‘long int’} [-Wformat=]
   15 |     printf( " GALAHAD version: %d.%d.%d\n", major, minor, patch );
      |                                ~^           ~~~~~
      |                                 |           |
      |                                 int         ipc_ {aka long int}
      |                                %ld
../src/version/C/versiont.c:15:36: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘ipc_’ {aka ‘long int’} [-Wformat=]
   15 |     printf( " GALAHAD version: %d.%d.%d\n", major, minor, patch );
      |                                   ~^               ~~~~~
      |                                    |               |
      |                                    int             ipc_ {aka long int}
      |                                   %ld
../src/version/C/versiont.c:15:39: warning: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘ipc_’ {aka ‘long int’} [-Wformat=]
   15 |     printf( " GALAHAD version: %d.%d.%d\n", major, minor, patch );
      |                                      ~^                   ~~~~~
      |                                       |                   |
      |                                       int                 ipc_ {aka long int}
      |                                      %ld
[3950/3966] Compiling C object versiont_c_quadruple.p/src_version_C_versiont.c.o
../src/version/C/versiont.c: In function ‘main’:
../src/version/C/versiont.c:15:33: warning: format ‘%d’ expects argument of type ‘int’, but argument 2 has type ‘ipc_’ {aka ‘long int’} [-Wformat=]
   15 |     printf( " GALAHAD version: %d.%d.%d\n", major, minor, patch );
      |                                ~^           ~~~~~
      |                                 |           |
      |                                 int         ipc_ {aka long int}
      |                                %ld
../src/version/C/versiont.c:15:36: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘ipc_’ {aka ‘long int’} [-Wformat=]
   15 |     printf( " GALAHAD version: %d.%d.%d\n", major, minor, patch );
      |                                   ~^               ~~~~~
      |                                    |               |
      |                                    int             ipc_ {aka long int}
      |                                   %ld
../src/version/C/versiont.c:15:39: warning: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘ipc_’ {aka ‘long int’} [-Wformat=]
   15 |     printf( " GALAHAD version: %d.%d.%d\n", major, minor, patch );
      |                                      ~^                   ~~~~~
      |                                       |                   |
      |                                       int                 ipc_ {aka long int}
      |                                      %ld
[3966/3966] Linking target wcptf_c_quadruple
```

--- a/.github/meson/action.yml
+++ b/.github/meson/action.yml
@@ -51,7 +51,7 @@ runs:
           echo "LIBDIR=lib" >> $GITHUB_ENV
           echo "DLEXT=so" >> $GITHUB_ENV
           echo "PLATFORM=x86_64-linux-gnu" >> $GITHUB_ENV
-          echo "JULIA_GALAHAD_LIBRARY_PATH=/scratch/github-actions/actions_runner_galahad/_work/GALAHAD/GALAHAD/GALAHAD.jl" >> $GITHUB_ENV
+          echo "JULIA_GALAHAD_LIBRARY_PATH=/scratch/github-actions/actions_runner_galahad/_work/GALAHAD/GALAHAD/galahad/lib" >> $GITHUB_ENV
         fi
         if [[ "${{ inputs.os }}" == "ubuntu-latest" ]]; then
           echo "LIBDIR=lib" >> $GITHUB_ENV

--- a/.github/meson/action.yml
+++ b/.github/meson/action.yml
@@ -51,7 +51,7 @@ runs:
           echo "LIBDIR=lib" >> $GITHUB_ENV
           echo "DLEXT=so" >> $GITHUB_ENV
           echo "PLATFORM=x86_64-linux-gnu" >> $GITHUB_ENV
-          echo "JULIA_GALAHAD_LIBRARY_PATH=$GITHUB_WORKSPACE/galahad/lib" >> $GITHUB_ENV
+          echo "JULIA_GALAHAD_LIBRARY_PATH=/scratch/github-actions/actions_runner_galahad/_work/GALAHAD/GALAHAD/GALAHAD.jl" >> $GITHUB_ENV
         fi
         if [[ "${{ inputs.os }}" == "ubuntu-latest" ]]; then
           echo "LIBDIR=lib" >> $GITHUB_ENV

--- a/include/galahad_expo.h
+++ b/include/galahad_expo.h
@@ -882,7 +882,7 @@ void expo_solve_hessian_direct( void **data,
                                 rpc_ gl[],
                                 ipc_ (*eval_fc)(
                                   ipc_, ipc_, const rpc_[],
-                                  rpc_, rpc_[], const void * ),
+                                  rpc_ *, rpc_[], const void * ),
                                 ipc_ (*eval_gj)(
                                   ipc_, ipc_, ipc_, const rpc_[],
                                   rpc_[], rpc_[], const void * ),

--- a/src/forthcoming/expo/Python/expo_pyiface.c
+++ b/src/forthcoming/expo/Python/expo_pyiface.c
@@ -48,7 +48,7 @@ static PyObject *py_eval_hl = NULL;
 static PyObject *expo_solve_return = NULL;
 
 /* C eval_* function wrappers */
-static int eval_fc(int n, int m, const double x[], double f, double c[],
+static int eval_fc(int n, int m, const double x[], double *f, double c[],
                    const void *userdata){
 
     // Wrap input array as NumPy array
@@ -69,7 +69,7 @@ static int eval_fc(int n, int m, const double x[], double f, double c[],
 
     // Extract eval_fc return values (one double and one array)
     PyObject *py_c;
-    if(!PyArg_ParseTuple(result, "dO", f, py_c)){
+    if(!PyArg_ParseTuple(result, "dO", f, &py_c)){
         PyErr_SetString(PyExc_TypeError,
         "unable to parse eval_fc return values");
         Py_DECREF(result); // Free result memory

--- a/src/forthcoming/expo/meson.build
+++ b/src/forthcoming/expo/meson.build
@@ -25,8 +25,8 @@ galahad_cutest_binaries += [['runexpo_sif', files('runexpo_sif.F90')]]
 galahad_tests += [# ['expo', 'expot', files('expot.F90')],
                   ['expo', 'expoti', files('expoti.F90')]]
 
-# galahad_c_tests += [['expo', 'expot_c', files('C/expot.c')],
-#                     ['expo', 'expotf_c', files('C/expotf.c')]]
+galahad_c_tests += [['expo', 'expot_c', files('C/expot.c')],
+                    ['expo', 'expotf_c', files('C/expotf.c')]]
 
 # galahad_python_tests += [['expo', 'expo_py', files('Python/test_expo.py')]]
 

--- a/src/version/C/versiont.c
+++ b/src/version/C/versiont.c
@@ -12,5 +12,5 @@ int main(void) {
     ipc_ patch;
 
     version_galahad( &major, &minor, &patch );
-    printf( " GALAHAD version: %d.%d.%d\n", major, minor, patch );
+    printf( " GALAHAD version: %1" i_ipc_ ".%1" i_ipc_ ".%1" i_ipc_ "\n", major, minor, patch );
 }

